templates/types/streaming/express/src/controllers/engine/settings.ts

```diff
@@ -85,6 +90,21 @@ function initAnthropic() {
   });
 }
 
+
+function initGroq() {
+  // To review
+  const embedModelMap: Record<string, string> = {
+    "all-MiniLM-L6-v2": "Xenova/all-MiniLM-L6-v2",
+    "all-mpnet-base-v2": "Xenova/all-mpnet-base-v2",
+  };
+  Settings.llm = new Groq({
+    model: process.env.MODEL as keyof typeof ALL_AVAILABLE_GROQ_MODELS,
```

Comment:
Needs modelMap lookup like in settings.py - values of the map need to be of type `keyof typeof ALL_AVAILABLE_GROQ_MODELS`
```suggestion
    model: modelMap[process.env.MODEL!],
```
(same also for nextjs)

Reply:
You mean doing something like this ? ```diff --git a/templates/types/streaming/nextjs/app/api/chat/engine/settings.ts b/templates/types/streaming/nextjs/app/api/chat/engine/settings.ts
index 48b0cac..3987cc1 100644
--- a/templates/types/streaming/nextjs/app/api/chat/engine/settings.ts
+++ b/templates/types/streaming/nextjs/app/api/chat/engine/settings.ts
@@ -79,9 +79,17 @@ function initGroq() {
     "all-MiniLM-L6-v2": "Xenova/all-MiniLM-L6-v2",
     "all-mpnet-base-v2": "Xenova/all-mpnet-base-v2",
   };
+
+  const modelMap: Record<string, string> = {
+        "llama3-8b": "llama3-8b-8192",
+        "llama3-70b": "llama3-70b-8192",
+        "mixtral-8x7b": "mixtral-8x7b-32768",
+  }
+
   Settings.llm = new Groq({
-    model: process.env.MODEL as keyof typeof ALL_AVAILABLE_GROQ_MODELS,
+    model: modelMap[process.env.MODEL!],
   });
+
   Settings.embedModel = new HuggingFaceEmbedding({
     modelType: embedModelMap[process.env.EMBEDDING_MODEL!],
   });
``` Tough honestly I'm not really sure how to do those things I just though of doing things similarly to what was done for anthropic but actually I haven't really understood this first line: ```import { ALL_AVAILABLE_ANTHROPIC_MODELS } from "llamaindex/llm/anthropic";
``` I mean I have never used typescript that is way I struggle a bit with that part

Reply:
@Jac-Zac yes exactly, that looks good. The `Groq` class supports `string` type as model, so don't worry about `ALL_AVAILABLE_GROQ_MODELS`.
This code should run, can you test it on TS?